packages/rxfs/src/operators/writeFile.js

```diff
@@ -9,12 +9,13 @@ const writeFile = function writeFile({
   _writeFile = bindNodeCallback(fsWrite),
 }) {
   return fileChunk$ => fileChunk$.pipe(
-    scan(([index, chunk]) => [index + 1, chunk], [-1, null]),
-    mergeMap(([index, chunk]) => (
```

Comment:
I'm pretty sure `mergeMap` is the same as `map, mergeAll(1)`

Reply:
You can also add concurrency to mergeMap as well.

Reply:
Oh!  So that might be a problem...  I'm pretty sure `mergeMap(doSomething, 1)` drops excess items.  I don't think it queues them.  If that's what mergeAll does, then maybe that's why my code is failing.  If mergeAll is already busy with a concurrent task, it disregards any tasks that it receives while the prior task is still running.  Since my fileWriter is asynchronous, maybe it receives the last two items but doesn't write them because they appear before the prior item has finished writing...

Reply:
That is totally possible. You might be wanting to use concatAll(), which is synchronous and queues items.